src/System.Private.CoreLib/src/Internal/Runtime/Augments/RuntimeThread.Unix.cs

```diff
@@ -2,62 +2,134 @@
 // The .NET Foundation licenses this file to you under the MIT license.
 // See the LICENSE file in the project root for more information.
 
+using Microsoft.Win32.SafeHandles;
 using System;
 using System.Diagnostics;
+using System.Runtime;
+using System.Runtime.InteropServices;
 using System.Threading;
 
 namespace Internal.Runtime.Augments
 {
     public sealed partial class RuntimeThread
     {
-        private readonly WaitSubsystem.ThreadWaitInfo _waitInfo;
+        private static bool s_setThreadExitCallback;
 
-        private void PlatformSpecificInitialize() { }
+        // Event signaling that the thread has stopped
+        private ManualResetEvent _stopped;
 
-        private void PlatformSpecificInitializeExistingThread() { }
+        private readonly WaitSubsystem.ThreadWaitInfo _waitInfo;
 
         internal WaitSubsystem.ThreadWaitInfo WaitInfo => _waitInfo;
 
-        private static ThreadPriority GetPriority() { return ThreadPriority.Normal; }
+        private void PlatformSpecificInitialize()
+        {
+            // Race condition is OK here since we set to the same value
+            if (!s_setThreadExitCallback)
```

Comment:
You can as well just set it all the time and get rid of the `s_setThreadExitCallback` flag.